File: Backend/app/schemas/meeting/meeting_events.py
from typing import Literal

from pydantic import BaseModel


class JobStarted(BaseModel):
    type: Literal["JobStarted"] = "JobStarted"
    total_meetings: int


class MeetingProcessingStarted(BaseModel):
    type: Literal["MeetingProcessingStarted"] = "MeetingProcessingStarted"
    index: int
    title: str


class MeetingProcessingSucceeded(BaseModel):
    type: Literal["MeetingProcessingSucceeded"] = "MeetingProcessingSucceeded"
    index: int
    title: str
    meeting_id: str


class MeetingProcessingFailed(BaseModel):
    type: Literal["MeetingProcessingError"] = "MeetingProcessingError"
    index: int
    title: str
    error: str


class JobFinished(BaseModel):
    type: Literal["JobFinished"] = "JobFinished"
    meetings_succeeded: int
    meetings_failed: int


MeetingUploadEvent = (
    JobStarted
    | MeetingProcessingStarted
    | MeetingProcessingSucceeded
    | MeetingProcessingFailed
    | JobFinished
)

File: Frontend/src/api/meetings/meetingEvents.ts
export interface JobStartedEvent {
  type: 'JobStarted'
  totalMeetings: number
}

export interface MeetingProcessingStartedEvent {
  type: 'MeetingProcessingStarted'
  index: number
  title: string
}

export interface MeetingProcessingSucceededEvent {
  type: 'MeetingProcessingSucceeded'
  index: number
  title: string
  meetingId: string
}

export interface MeetingProcessingFailedEvent {
  type: 'MeetingProcessingFailed'
  index: number
  title: string
  error: string
}

export interface MeetingProcessingErrorEvent {
  type: 'MeetingProcessingError'
  index: number
  title: string
  error: string
}

export interface JobFinishedEvent {
  type: 'JobFinished'
  meetingsSucceeded: number
  meetingsFailed: number
}

export type MeetingUploadEvent =
  | JobStartedEvent
  | MeetingProcessingStartedEvent
  | MeetingProcessingSucceededEvent
  | MeetingProcessingFailedEvent
  | MeetingProcessingErrorEvent
  | JobFinishedEvent
